File: backend/src/main/java/com/back/ourlog/domain/tag/entity/Tag.java
package com.back.ourlog.domain.tag.entity;

import jakarta.persistence.*;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Tag {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true)
    private String name;

    @OneToMany(mappedBy = "tag", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DiaryTag> diaryTags = new ArrayList<>();

    public Tag(String name) {
        this.name = name;
    }

}

File: backend/src/main/java/com/back/ourlog/domain/tag/entity/Tag.kt
package com.back.ourlog.domain.tag.entity

import jakarta.persistence.*
import org.hibernate.annotations.BatchSize

@Entity
@BatchSize(size = 50)
class Tag(
    @Column(unique = true, nullable = false)
    var name: String
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0

    @OneToMany(mappedBy = "tag", cascade = [CascadeType.ALL], orphanRemoval = true)
    val diaryTags: MutableList<DiaryTag> = mutableListOf()

    protected constructor() : this("")
}
